data_portal/models.py

```diff
@@ -544,6 +544,36 @@ def from_seq_run_status(cls, value):
             raise ValueError(f"No matching SequenceStatus found for value: {value}")
 
 
+class SequenceManager(models.Manager):
+
+    def get_by_keyword(self, column_names, **kwargs) -> QuerySet:
```

Comment:
Right! I appreciate the creativity here, Will. But, let not do this. 

This `column_names` effectively invert giving control on Model structure to external. In this case, those calls from viewset controller layer. This will defer logic into data access layer. And, we will need to handle more control logic, testing and logging, etc... eventually.

General guideline here is, we should keep these data access managers slime, small and, only query and data retrieval related logic. Things like those common queries needed by controller or service layer. Most of all, we should not let external layer control the Model structure. Managers are DAO and they are closet akin to Model; and hence, it should protect that as well.

Having said, I personally like the code snippet here. Though I'd use it in some simpler project with just couple of Models and, just need to REST/CRUD them!

Reply:
Yes, I thought it was too redundant to redeclare the same thing over and over so was thinking of putting it into an array and for a loop.

Wondering if we should bring the column array to this function and control which column to filter from here or would be better to use the old pattern?

Reply:
TL;DR, pls remove "column array" and, pls follow other manager impl for keywords lookup.

There shouldn't be "column array" because we define this in Model. When model make changes on this, this "column array" will need to align up or, else code break. 

kwargs is dict and, just need to do a lookup. Lookup is expressive and safe by design/nature. i.e. we do not need to explicitly test it or if-else guard checking it. Most of all, if model changes this attribute, it won't crash the program but, just return possibly a blank or not found item from database.

Reply:
Unless you use reflection to introspect the object to get all attributes. Oh well! Yah you can do that. See if that work...